src/nlp/api.jl

```diff
@@ -775,7 +773,9 @@ function jac_lin_op!(
     return res
   end
   ctprod! = @closure (res, v, α, β) -> begin
-    jtprod_lin!(nlp, x, v, Jtv)
+    rows, cols = jac_lin_structure(nlp)
+    vals = jac_lin_coord(nlp)
+    jtprod_lin!(nlp, rows, cols, vals, v, Jtv)
     if β == 0
       res .= α .* Jtv
     else
```

Comment:
Keep the old jac_lin_op!, add a line in the docstring to say it's deprecated. Use the macro @deprecate.